tf_quant_finance/black_scholes/vanilla_prices_test.py

```diff
@@ -334,6 +334,144 @@ def test_binary_vanilla_consistency_exact(self):
+    continuous_dividends = 0.04
+    exp = self.get_barrier_option_test_vals("pui")
+    strikes = exp[0]
+    barriers = exp[1]
+    expected_price = exp[2]
+    is_call_options = exp[3]
+    is_barrier_down = exp[4]
+    is_knock_out = exp[5]
+    volatilities = 0.25
+
+    """
+    3 -> cdi
+    2 -> pdi
+    1 -> cui
+    0 -> pui
+    7 -> cdo
+    6 -> pdo
+    5 -> cuo
+    4 -> puo
+    """
+    price = tff.black_scholes.barrier_price(
+        volatilities=volatilities, strikes=strikes,
+        expiries=expiries, spots=spots,
+        discount_rates=discount_rates,
+        continuous_dividends=continuous_dividends,
+        barriers=barriers, rebates=rebates,
+        is_barrier_down=is_barrier_down,
+        is_knock_out=is_knock_out,
+        is_call_options=is_call_options)
+    self.assertAllClose(price, expected_price, 10e-3)
+
+  def get_barrier_option_test_vals(self, param):
+    """Function returns testing vals for type of option"""
+    if param == "cdo":
+      return (90, 95, 9.0246, True, True, True, 7)
+    if param == "cdi":
+      return (90, 95, 7.7627, True, True, False, 3)
+    if param == "cuo":
+      return (90, 105, 2.6789, True, False, True, 5)
+    if param == "cui":
+      return (90, 105, 14.1112, True, False, False, 1)
+    if param == "pdo":
+      return (90, 95, 2.2798, False, True, True, 6)
+    if param == "puo":
+      return (90, 105, 3.7760, False, False, True, 4)
+    if param == "pdi":
+      return (90, 95, 2.9586, False, True, False, 2)
+    if param == "pui":
+      return (90, 105, 1.4653, False, False, False, 0)
+
+  def test_barrier_option_2d(self):
+    """Function tests barrier option pricing for vector inputs"""
+    spots = [100., 100., 100., 100., 100., 100., 100., 100.]
+    rebates = [3., 3., 3., 3., 3., 3., 3., 3.]
+    expiries = [.5, .5, .5, .5, .5, .5, .5, .5]
+    discount_rates = [.08, .08, .08, .08, .08, .08, .08, .08]
+    volatilities = [.25, .25, .25, .25, .25, .25, .25, .25]
+    strikes = [90., 90., 90., 90., 90., 90., 90., 90.]
+    barriers = [95., 95., 105., 105., 95., 105., 95., 105.]
+    expected_price = [
+        9.024, 7.7627, 2.6789, 14.1112, 2.2798, 3.7760, 2.95586, 1.4653]
+    is_call_options = [True, True, True, True, False, False, False, False]
+    is_barrier_down = [True, True, False, False, True, False, True, False]
+    is_knock_out =    [True, False, True, False, True, True, False, False]
+    continuous_dividends = [.04, .04, .04, .04, .04, .04, .04, .04]
+    price = tff.black_scholes.barrier_price(
+        volatilities=volatilities, strikes=strikes,
+        expiries=expiries, spots=spots,
+        discount_rates=discount_rates,
+        continuous_dividends=continuous_dividends,
+        barriers=barriers, rebates=rebates,
+        is_barrier_down=is_barrier_down,
+        is_knock_out=is_knock_out,
+        is_call_options=is_call_options)
+    self.assertAllClose(price, expected_price, 10e-3)
+
+  def test_barrier_option_cost_of_carriers(self):
```

Comment:
Could you please add one more test to ensure XLA-compatability? Please refer to [this](https://github.com/google/tf-quant-finance/blob/1239ce3a22b4a6545d1600835333e8c9984ade7d/tf_quant_finance/experimental/lsm_algorithm/lsm_v2_test.py#L114) example. Let me know if you have any issues with setting it up.